src/indexer/segment_writer.rs

```diff
@@ -157,7 +158,11 @@ impl SegmentWriter {
 
     fn index_document(&mut self, doc: &Document) -> crate::Result<()> {
         let doc_id = self.max_doc;
-        for (field, values) in doc.get_sorted_field_values() {
+        let mut field_vals: Vec<_> = doc.field_values().iter().collect();
+        field_vals.sort_by_key(|el| el.field());
+        let iter = field_vals.into_iter().group_by(|el| el.field());
+        for (field, field_values) in &iter {
```

Comment:
can it be 

```
for (field, field_values) in iter {
}
```

Reply:
no, that's some weirdness of the group_by api